internal/scm/gitlab/gitlab.go

```diff
@@ -385,15 +390,94 @@ func (g *Gitlab) MergePullRequest(ctx context.Context, pullReq domain.PullReques
 func (g *Gitlab) ClosePullRequest(ctx context.Context, pullReq domain.PullRequest) error {
 	pr := pullReq.(pullRequest)
 
-	_, err := g.glClient.MergeRequests.DeleteMergeRequest(pr.pid, pr.iid, gitlab.WithContext(ctx))
+	_, err := g.glClient.MergeRequests.DeleteMergeRequest(pr.targetPID, pr.iid, gitlab.WithContext(ctx))
 	if err != nil {
 		return err
 	}
 
-	_, err = g.glClient.Branches.DeleteBranch(pr.pid, pr.branchName, gitlab.WithContext(ctx))
+	_, err = g.glClient.Branches.DeleteBranch(pr.sourcePID, pr.branchName, gitlab.WithContext(ctx))
 	if err != nil {
 		return err
 	}
 
 	return nil
 }
+
+// ForkRepository forks a project
+func (g *Gitlab) ForkRepository(ctx context.Context, repo domain.Repository, newOwner string) (domain.Repository, error) {
+	r := repo.(repository)
+
+	// Get the username of the fork (logged in user if none is set)
+	ownerUsername := newOwner
+	if newOwner == "" {
+		currentUser, err := g.getCurrentUser(ctx)
+		if err != nil {
+			return nil, err
+		}
+		ownerUsername = currentUser.Username
+	}
+
+	// Check if the project already exist
+	project, resp, err := g.glClient.Projects.GetProject(
+		fmt.Sprintf("%s/%s", ownerUsername, r.name),
+		nil,
+		gitlab.WithContext(ctx),
+	)
+	if err == nil { // Already forked, just return it
+		return convertProject(project)
+	} else if resp.StatusCode != http.StatusNotFound { // If the error was that the project does not exist, continue to fork it
+		return nil, err
+	}
+
+	newRepo, _, err := g.glClient.Projects.ForkProject(r.pid, &gitlab.ForkProjectOptions{
+		Namespace: &newOwner,
+	}, gitlab.WithContext(ctx))
+	if err != nil {
+		return nil, err
+	}
+
+	for i := 0; i < 10; i++ {
+		repo, _, err := g.glClient.Projects.GetProject(newRepo.ID, nil, gitlab.WithContext(ctx))
+		if err != nil {
+			return nil, err
+		}
+
+		if repo.ImportStatus == "finished" {
+			return convertProject(newRepo)
+		}
+
+		time.Sleep(time.Second * 3)
+	}
+
+	return nil, errors.New("time waiting for fork to complete was exceeded")
```

Comment:
we probably should standardize the error message between the different services? This doesn't match what is reported when forking from GitHub takes too long.